### DISCUSSION TITLE
Improve docs for testing with Pest

### DISCUSSION THREAD

**Discussion by back2Lobby** [nWidart/laravel-modules#1934] [General]
I was reading the [docs](https://laravelmodules.com/docs/v11/tests#use-pestphp) for tests pest. It suggests the users to add the `uses(Tests\TestCase::class);` at the top of every single file. But I didn't like it and started digging for better solution. Here is what I found.

According to pest [docs](https://pestphp.com/docs/configuring-tests#:~:text=Another%20more%20complex%20example%20would%20be%20using%20a%20pattern%20to%20match%20multiple%20directories%20in%20different%20modules%20while%20applying%20multiple%20test%20case%20classes%20and%20traits%3A), you only need to add this in the `tests/Pest.php`:
```php
uses(
    TestCase::class,
    RefreshDatabase::class
)->in('../Modules/*/Tests/Feature');
```
This way we don't need to include the `uses` method in every single file. So, I guess this should be included in the docs, what do you think?

**Comment by dcblogdev** [comment]
yep, that works just as long as you remove `uses(TestCase::class);` from any existing module tests files.

To have test work for any test file this works:

```php
uses(
    TestCase::class,
)->in('../Modules/*/Tests/*');
```

**Reply by dcblogdev** [reply]
Definitely worth a docs entry.

**Comment by dcblogdev** [comment]
converted this as an issue to remind myself to action this.

https://github.com/nWidart/laravel-modules/issues/1936
